core/src/main/java/com/datastax/oss/driver/internal/core/CqlIdentifiers.java

```diff
@@ -16,21 +16,36 @@
 package com.datastax.oss.driver.internal.core;
 
 import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.shaded.guava.common.base.Preconditions;
 import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
 import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableMap;
+import edu.umd.cs.findbugs.annotations.NonNull;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
 
 public class CqlIdentifiers {
 
-  public static List<CqlIdentifier> wrap(Iterable<String> in) {
+  public static List<CqlIdentifier> wrap(
+      @NonNull Iterable<String> in, @NonNull Function<String, CqlIdentifier> fn) {
+
+    Preconditions.checkNotNull(in, "Input Iterable must not be null");
+    Preconditions.checkNotNull(fn, "CqlIdentifier conversion function must not be null");
     ImmutableList.Builder<CqlIdentifier> out = ImmutableList.builder();
     for (String name : in) {
-      out.add(CqlIdentifier.fromCql(name));
+      out.add(fn.apply(name));
     }
     return out.build();
   }
 
+  public static List<CqlIdentifier> wrap(Iterable<String> in) {
```

Comment:
I would suggest that you rename this method to `wrapCql`. It's an internal API so we don't care about backwards compatibility.

Reply:
That actually was my preference... but I was concerned about the API modification.  Good to know that's more relaxed for the internal stuff (which does make sense).

Reply:
Sure, but those costs have to be paid anytime you decide you want to rename some methods, so it seems like the real question here is whether the proposed names are more descriptive.  And it does seem to me like they are, especially in light of the other changes being made here.

Reply:
I'd argue that a more coherent naming scheme is a pretty reasonable argument for an update under the criteria laid out above... but I don't feel strongly enough about it to go to the mat.  And I'd rather wind up with a clear API for developers to use rather than leave some deprecated methods around.  So I'll defer and remove the rename.